de.metas.procurement.base/src/main/java/de/metas/procurement/base/IPMMContractsDAO.java

```diff
@@ -46,4 +46,15 @@
 
 	boolean hasRunningContracts(I_PMM_Product pmmProduct);
 
+	/**
+	 * Retrieve the running procurement contract for the given Date, Partner and Product.
+	 * In case there are many, choose the one with the latest start date
+	 * 
```

Comment:
..but that start date may not be *after* <code>date</code>, right. having scrolled down, i think that's missing in the implementation